std/file.d

```diff
@@ -946,13 +947,96 @@ if (isConvertibleToString!RF || isConvertibleToString!RT)
     assert(t2.readText == "2");
 }
 
-private void renameImpl(const(char)[] f, const(char)[] t, const(FSChar)* fromz, const(FSChar)* toz) @trusted
+@safe unittest
+{
+    import std.utf : byWchar;
+
+    auto t1 = deleteme, t2 = deleteme~"2";
+    scope(exit) foreach (t; [t1, t2]) if (t.exists) t.remove();
+
+    write(t1, "1");
+    rename(t1, t2);
+    assert(readText(t2) == "1");
+
+    write(t1, "2");
+    rename(t1, t2.byWchar);
+    assert(readText(t2) == "2");
+}
+
+
+/***************************************************
+Move file `from` _to `to`.
+If the target file exists, it is overwritten. Unlike `rename` this function supports
+moving files accross drives/filesystems.
+Params:
+   from = string or range of characters representing the existing file name
+   to = string or range of characters representing the target file name
+Throws: `FileException` on error.
+*/
+void moveFile(RF, RT)(RF from, RT to)
+if ((isInputRange!RF && !isInfinite!RF && isSomeChar!(ElementEncodingType!RF) || isSomeString!RF)
+    && !isConvertibleToString!RF &&
+    (isInputRange!RT && !isInfinite!RT && isSomeChar!(ElementEncodingType!RT) || isSomeString!RT)
+    && !isConvertibleToString!RT)
+{
+    // Place outside of @trusted block
+    auto fromz = from.tempCString!FSChar();
+    auto toz = to.tempCString!FSChar();
+
+    static if (isNarrowString!RF && is(Unqual!(ElementEncodingType!RF) == char))
+        alias f = from;
+    else
+        enum string f = null;
+
+    static if (isNarrowString!RT && is(Unqual!(ElementEncodingType!RT) == char))
+        alias t = to;
+    else
+        enum string t = null;
+
+    moveFileImpl(f, t, fromz, toz, Yes.canCopy);
+}
+
+/// ditto
+void moveFile(RF, RT)(auto ref RF from, auto ref RT to)
+if (isConvertibleToString!RF || isConvertibleToString!RT)
```

Comment:
IIRC this overload was only added for legacy compatibility. So I'm not sure whether new symbols need it.
Or in other words: do we want to support aliased-strings for this new symbol.

Reply:
IMHO, it's a mistake to add any new code that supports such implicit conversions, and either way, `isConvertibleToString` should not be used in any new code. It's going to be deprecated once Phobos doesn't use it anymore, because it's generally unsafe (though this particular case doesn't suffer from that problem).